conans/test/integration/package_id/test_cache_compatibles.py

```diff
@@ -236,3 +236,131 @@ class Pkg(ConanFile):
+        client.assert_listed_binary({"sum/0.1": ("da39a3ee5e6b4b0d3255bfef95601890afd80709",
+                                                 "Build")})
+        # Just check that it works and doesn't fail
+        assert "Installing (downloading, building) binaries" in client.out
+        client.run("create . -s compiler.cppstd=14 --build missing")
+        # Now it will not build, as package exist
+        client.assert_listed_binary({"sum/0.1": ("da39a3ee5e6b4b0d3255bfef95601890afd80709",
+                                                 "Cache")})
+        assert "Installing (downloading, building) binaries" in client.out
+
+    def test_check_min_cppstd(self):
+        """ test that the check_min_cppstd works fine wiht compatibility, as it is based
+        on ``conanfile.info.settings`` not ``conanfile.settings``
+        """
+        conanfile = textwrap.dedent("""
+            from conan import ConanFile
+            from conan.tools.build import check_min_cppstd, valid_min_cppstd
+            class Pkg(ConanFile):
+                name = "pkg"
+                version = "0.1"
+                settings = "os", "arch", "compiler", "build_type"
+                def validate(self):
+                    check_min_cppstd(self, "17", False)
+                    self.output.info("valid standard!!")
+                def package_info(self):
+                    self.output.info("CPPSTD: {}".format(self.settings.compiler.cppstd))
+            """)
+
+        c = TestClient()
+        c.save({"conanfile.py": conanfile})
+        settings = "-s compiler=gcc -s compiler.version=9 -s compiler.libcxx=libstdc++11"
+        c.run(f"create .  {settings} -s compiler.cppstd=17")
+        assert "pkg/0.1: valid standard!!" in c.out
+        assert "pkg/0.1: CPPSTD: 17" in c.out
+        c.run(f"install {settings} --requires=pkg/0.1 -s compiler.cppstd=14")
+        assert "valid standard!!" in c.out
+        assert "pkg/0.1: CPPSTD: 17" in c.out
+        c.run(f"install {settings} --requires=pkg/0.1 -s compiler.cppstd=20")
+        assert "valid standard!!" in c.out
+        assert "pkg/0.1: CPPSTD: 17" in c.out
+
+    def test_check_min_cppstd_interface(self):
+        """ test that says that compatible binaries are ok, as long as the user defined
+        cppstd>=14. The syntax is a bit forced, maybe we want to improve ``check_min_cppstd``
+        capabilities to be able to raise ConanErrorConfiguration too
+        """
+        conanfile = textwrap.dedent("""
+            from conan import ConanFile
+            from conan.errors import ConanErrorConfiguration
+            from conan.tools.build import check_min_cppstd, valid_min_cppstd
+            class Pkg(ConanFile):
+                name = "pkg"
+                version = "0.1"
+                settings = "os", "arch", "compiler", "build_type"
+                def validate(self):
+                    if int(str(self.info.settings.compiler.cppstd).replace("gnu", "")) <= 14:
```

Comment:
Building this validation atm (for hard ConfigurationError, not fallback to other compatible binary) is ugly, but feasible. We can improve ``check_min_cppstd()`` helper later.